toolchest_client/tools/tool_args.py

```diff
@@ -0,0 +1,326 @@
+"""
+toolchest_client.tools.arg_whitelist
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This contains lists of custom tool_args that can currently be
+processed by Toolchest.
+
+The whitelist is a dict of dicts for each tool, with each tool dict
+containing accepted tags as keys and each tag's required number of
+following args as the keys' vals.
+
+The blacklist is a list of arguments that we know are not compatible
+with Toolchest. A tag that is found in this list will cause the run
+to fail.
+
+The dangerlist is a list of arguments that we know will change
+the function or structure of the tool. A tag that is found in this
+list will cause the run to reduce complexity and validation (e.g.
+not parallelization and less validation).
+
+Note: Some tags (e.g., for inputs: "-1", "-2") are filtered out, but
+their functionalities are provided via other arguments (e.g.,
+user-provided inputs).
+"""
+
+VARIABLE_ARGS = "variable"
+
+TOOL_ARG_LISTS = {
```

Comment:
These are mostly the same args as before, just structured as an explicit whitelist / dangerlist / blacklist.